### PR TITLE
Mongodb memory connector initial

### PR DESCRIPTION
**Pull request by BorisDog**
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/MongoDbTest/Program.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.SemanticKernel.Connectors.Memory.MongoDB;
+using Microsoft.SemanticKernel.AI.Embeddings.VectorOperations;
+
+Console.WriteLine("Hello, World!");
+
+const string connectionUri = "mongodb+srv://boris:Pas12345@cluster0.mqddkqg.mongodb.net/?retryWrites=true&w=majority";
```
Comment (JamesKovacs) [inline dotnet/MongoDbTest/Program.cs]: Atlas credentials embedded in test code.

File: dotnet/MongoDbTest/Program.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.SemanticKernel.Connectors.Memory.MongoDB;
+using Microsoft.SemanticKernel.AI.Embeddings.VectorOperations;
+
+Console.WriteLine("Hello, World!");
```
Comment (JamesKovacs) [inline dotnet/MongoDbTest/Program.cs]: Shouldn't this say something like "Starting test..." or similar?

File: dotnet/MongoDbTest/Program.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+var f = v.Select(f => (float)f).ToArray();
```
Comment (JamesKovacs) [inline dotnet/MongoDbTest/Program.cs]: Why is the vector declared as an array of double but then the values cast to floats?

File: dotnet/MongoDbTest/Program.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+var r = await store.GetNearestMatchAsync("embedded_movies_sk", f, withEmbedding: true);
+
+double similarity = r.Value.Item1.Embedding
+    .Span
+    .CosineSimilarity(f);
+
+double similarity2 = r.Value.Item1.Embedding
+    .Span
+    .CosineSimilarity(f);
```
Comment (JamesKovacs) [inline dotnet/MongoDbTest/Program.cs]: `similarity` and `similarity2` are identical. Why is it called twice?

File: dotnet/samples/KernelSyntaxExamples/Example15_TextMemorySkill.cs [hunk 5]
```diff
@@ -44,6 +45,9 @@ public static async Task RunAsync(CancellationToken cancellationToken = default)
         // DuckDB Memory Store - a file-based store that persists data in a DuckDB database
         // store = await CreateSampleDuckDbMemoryStoreAsync();
 
+        // MongoDB Memory Store - a store that persists data in a MongoDB database
+        //store = CreateSampleMongoDBMemoryStore();
```
Comment (JamesKovacs) [inline dotnet/samples/KernelSyntaxExamples/Example15_TextMemorySkill.cs]: Missing a space between `//` and `store`.

File: dotnet/src/Connectors/Connectors.Memory.MongoDB/MongoDBMemoryEntry.cs [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MongoDBMemoryEntry"/> class.
+    /// </summary>
+    /// <param name="memoryRecord"><see cref="MemoryRecord"/>Instance to copy values from.</param>
+    public MongoDBMemoryEntry(MemoryRecord memoryRecord)
+    {
+        this.Id = memoryRecord.Key;
+        this.Metadata = memoryRecord.Metadata; // Metadata is readonly
+        this.Embedding = memoryRecord.Embedding.ToArray();
```
Comment (JamesKovacs) [inline dotnet/src/Connectors/Connectors.Memory.MongoDB/MongoDBMemoryEntry.cs:56]: Is `MemoryRecord.Embedding` guaranteed not to be null? I checked the docs and it doesn't say either way.

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
This looks fantastic! Minor feedback inline.

File: dotnet/MongoDbTest/Program.cs (hunk 2 above)
Comment (BorisDog) [reply dotnet/MongoDbTest/Program.cs]: Oh it's a leftover project I forgot to clean up.
Removed MongoDbTest. Thanks!

File: dotnet/samples/KernelSyntaxExamples/Example15_TextMemorySkill.cs (hunk 5 above)
Comment (BorisDog) [reply dotnet/samples/KernelSyntaxExamples/Example15_TextMemorySkill.cs]: Tnx, done.

File: dotnet/src/Connectors/Connectors.Memory.MongoDB/MongoDBMemoryEntry.cs (hunk 6 above)
Comment (BorisDog) [reply dotnet/src/Connectors/Connectors.Memory.MongoDB/MongoDBMemoryEntry.cs:56]: Yes, it's type is `ReadonlyMemory<float>`, which is a struct.

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request closed without merging**